content/docs/document-lifecycle.md

```diff
@@ -1,8 +1,10 @@
 ---
 title: 'Document Lifecycle'
-weight: 300
+weight: 400
 ---
 
+`LangiumDocument` is the central data structure in Langium that represents a text file of your DSL. Its main purpose is to hold the parsed Abstract Syntax Tree (AST) plus additional information derived from it. After its creation, a `LangiumDocument` must be "built" before it can be used in any way. The service responsible for building documents is called `DocumentBuilder`.
```

Comment:
Might be a bit too much of a niche for this part of the documentation but should we mention that `LangiumDocument` can also be generated programatically and don't have to necessarily come from a text file?